code/modules/modular_computers/file_system/data.dm

```diff
@@ -7,10 +7,12 @@
 	var/do_not_edit = 0				// Whether the user will be reminded that the file probably shouldn't be edited.
 	var/read_only = 0				// Protects files that should never be edited by the user due to special properties.
 
-/datum/computer_file/data/clone()
-	var/datum/computer_file/data/temp = ..()
-	temp.stored_data = stored_data
-	return temp
+/datum/computer_file/data/Clone(datum/computer_file/data/copy_instance = null, rename = FALSE)
+	if(!copy_instance)
+		copy_instance = new type
```

Comment:
Because this is more flexible. You can pass any type you want down the chain. So say you want your copies to be of another type that inherits from the same parents, you can just create it from the type you want, and pass it down. 
That could be handy for things like contracts that would all keep a reference to the source contract and cancel themselves when the master contract is cancelled, and stuff like that.

Reply:
Also, constructor parameters are validated in many atoms, so you have to be able to override the creation of the instance per subclass.